File: lecture_6/assignements/sentimental-mario-less/mario.py
from cs50 import get_int


def main():
    height = get_valid_height()
    print_pyramid(height)


def print_pyramid(n):
    for i in range(n):
        print(" " * (n - (i + 1)), end="")
        print("#" * (i + 1))


def get_valid_height():
    while True:
        height = get_int("Height: ")

        if 1 <= height <= 8:
            return height


main()

File: lecture_0-2/assignements/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void print_row(int spaces, int bricks);

int main(void)
{
     // Prompt the user for the pyramid's height
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1);

    // Print a pyramid of that height
    for (int i = 0; i < height; i++)
    {
        // Print the first half of a pyramid row
        print_row(height - i + 1, i + 1);

        // Print the middle space
        printf("  ");

        // Print the second half of a pyramid row
        print_row(0, i + 1);

        // Go to the next row
        printf("\n");
    }
}

void print_row(int spaces, int bricks)
{
    // Print spaces (decreasing)
    for (int i = spaces; i > 2; i--)
    {
        printf(" ");
    }

    // Print bricks (increasing)
    for (int j = 0; j < bricks; j++)
    {
        printf("#");
    }
}
